ksql-engine/src/main/java/io/confluent/ksql/materialization/Materialization.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.materialization;
+
+import io.confluent.ksql.model.WindowType;
+import io.confluent.ksql.schema.ksql.LogicalSchema;
+import java.util.Optional;
+
+public interface Materialization {
+
+  /**
+   * @return The schema of the materialization
+   */
+  LogicalSchema schema();
+
+  /**
+   * @return service for locating which node holds specific data.
+   */
+  Locator locator();
+
+  /**
+   * @return the window type of the materialization.
+   */
+  Optional<WindowType> windowType();
```

Comment:
I'm wanting to completely restructure how static queries are handled. As I said in the description, I'm planning on making use of `QueryAnalyser`, `LogicalPlanner` etc.

My thinking is that the initial analysis of static queries shouldn't be any different that non-static. It's possible even the logical plan steps would remain the same.  It should only be the physical planner that changes.  It should look for materializations of the data sources required and, where available, make use of them rather than using KS topologies.

This is my target architecture. But deadlines being tight, and with @rodesai doing a lot of restructuring around the logical and physical planners... I'm focusing more on functionality.

I agree this API isn't right. What I think we need is to have a common interface between windowed and non-windowed tables. Then we pass the WHERE clauses to them and they then determine if they have enough bounds to do their job, and what rows they're going to return, i.e. we inverse the responsibility.

It'll get there!